File: Leetcode-Medium/48 - Rotate Image/RotateImage.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        temp1 = []
        j=0
        while j<len(matrix[0]):
            for l in range(len(matrix)-1,-1,-1):
                item = matrix[l]
                temp1.append(item[j])
            j+=1
        k=0
        leng = len(matrix[0])
        for i in range(len(matrix)):
            matrix[i] = temp1[k:k+leng]
            k+=leng

File: Leetcode-Medium/48 - Rotate Image/Rotate Image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        vector<int>v;
        int s=matrix.size();
        for(int i=matrix.size()-1;i>=0;i--){
            for(int j=0;j<matrix[i].size();j++){
                v.push_back(matrix[i][j]);
            }
        }
        int k=0;
        for(int j=0;j<matrix.size();j++){
            for(int i=0;i<matrix[j].size();i++){
                matrix[i][j]=v[k];
                k++;
            }
        }
    }
};
